client/packages/common/src/plugins/types.ts

```diff
@@ -1,68 +1,23 @@
-import { FunctionComponent } from 'react';
-import { ColumnDefinition, RecordWithId } from '@openmsupply-client/common';
+import { ColumnDefinition, UsePluginEvents } from '@openmsupply-client/common';
 import { StockLineRowFragment } from '@openmsupply-client/system';
 import { InboundFragment } from '@openmsupply-client/invoices';
 
-type extractDataType<Type> = Type extends ComponentPluginBase<infer DataType>
-  ? DataType
-  : never;
-
-export type ComponentPluginData<T> = extractDataType<
-  Extract<ComponentPlugin, { type: T }>
->;
-
-export type PluginComponent<T> = FunctionComponent<{ data: T }>;
-
-export type PluginModule<DataType> = {
-  default: React.ComponentType<{ data?: DataType | undefined }>;
+export type Plugins = {
```

Comment:
Simplified types and easier to add new interface, and it's implementation in plugin